HomeTask/HomeTask/Model/CoreDataManager.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  TasksPersistence.swift
+//  HomeTask
+//
+//  Created by Lucas Migge de Barros on 11/01/23.
+//
+
+import Foundation
+import CoreData
+
+
+protocol TaskPersistence {
+    func fetchTaks(completion: @escaping ([TaskItem]) -> Void)
+    
+    func saveTask(taskTitle: String, completion: @escaping () -> Void)
+    
+    func deleteTask(forTask task: TaskItem, completion: @escaping () -> Void)
+}
+
+
+class CoreDataManager: TaskPersistence {
+    
+    static var shared = CoreDataManager()
+    
+    let container: NSPersistentContainer
+    
+    var viewContext: NSManagedObjectContext {
+        container.viewContext
+    }
+    
+   
+    init() {
+        container = NSPersistentContainer(name: "TaskItem")
+        container.loadPersistentStores { description, error in
+            if let error = error {
+                print("Core Data failed to load: \(error.localizedDescription)")
+            }
+        }
+    }
```

Comment:
Mesma coisa sobre injeção de dependência. Nesse caso, é importante que `container` seja injetado nessa classe, ao invés de inicializado por ela mesmo. Isso nos permitiria rodas os testes em cima de versões diferentes desse container, de forma a poder reproduzir diferentes cenários e não afetar os dados reais do app.
```suggestion
    let container: NSPersistentContainer
    
    var viewContext: NSManagedObjectContext {
        container.viewContext
    }
    
   
    init(
        persistentContainer: NSPersistenceContainer = NSPersistentContainer(name: "TaskItem")
    ) {
        container = persistentContainer
        container.loadPersistentStores { description, error in
            if let error = error {
                print("Core Data failed to load: \(error.localizedDescription)")
            }
        }
    }
```

Reply:
Topaaado. Tu tem uma visão absurrrda. Mais uma vez, animado pra aula de testessss